dataplane/apigen/docparser/docparser.go

```diff
@@ -171,62 +196,138 @@ func memberToAttrInfo(enum MemberDef) *Attr {
 			info.SetFields = append(info.SetFields, atn)
 		}
 	}
-	return info
+	return info, nil
 }
 
-func memberToEnumValueStrings(enum MemberDef) []*Enum {
-	res := []*Enum{}
-	prev := -1 // Since enum values may repeat, store previous enum value. If the enum value isn't an assignment, then it is prev + 1.
-	for i, value := range enum.EnumValues {
-		val := prev + 1
-		value.Initializer = strings.TrimSpace(strings.TrimPrefix(value.Initializer, "= "))
-		if value.Initializer != "" {
-			if v, err := strconv.ParseUint(value.Initializer, 0, 64); err == nil {
-				val = int(v)
+var saiConsts = map[string]int{
+	"SAI_ACL_USER_DEFINED_FIELD_ATTR_ID_RANGE": 0xff,
+}
+
+func parseInitializer(i int, vals []EnumValue) (val int, rerr error) {
+	defer func() {
+		if rerr == nil {
+			saiConsts[vals[i].Name] = val
+		}
+	}()
+
+	findIndex := func(val string, vals []EnumValue) int {
+		for i, eV := range vals {
+			if eV.Name == val {
+				return i
 			}
-			for j := 0; j < i; j++ {
-				if value.Initializer == enum.EnumValues[j].Name {
-					val = res[j].Value
-					break
-				}
+		}
+		return -1
+	}
+
+	init := strings.TrimSpace(strings.TrimPrefix(vals[i].Initializer, "= "))
+	if init == "" {
+		if i == 0 {
+			return 0, nil
+		}
+		prev, err := parseInitializer(i-1, vals)
+		return prev + 1, err
+	}
+	if strings.Contains(init, "+") {
+		splits := strings.Split(init, "+")
```

Comment:
Do we want to assert splits has exactly 2 elements?

Reply:
done